mitodl/open-discussions
# Add all prolearn imports to professional offerings, handle multiple runs

#### Pre-Flight checklist



#### What are the relevant tickets?
Closes #3795 
Closes #3796

#### What's this PR do?
- Adds all courses/programs imported via the prolearn API to "Professional Offerings"
- If a course/programs has multiple start/end dates, create a run for each.

#### How should this be manually tested?
- Set `PROLEARN_CATALOG_API_URL=https://prolearn.mit.edu/graphql` in your .env file
- If you've already imported these before, delete them:
    ```
     manage.py backpopulate_prolearn_data --delete
    ```
- Otherwise or afterward, run the above without `--delete` to import them.
- Go to `../learn/search` and check the `Professional Offerings` checkbox.  Under "Offered By" you should see the following, plus XPro if you've imported anything from there:

![Screen Shot 2023-01-09 at 10 05 25 AM](https://user-images.githubusercontent.com/187676/211339979-74ec8e73-e524-42c2-ad3e-5f30aa2312e2.png)

- Search for "accelerating digital", click on the "Accelerating Digital Transformation with Algorithmic Business Thinking" course to bring up the drawer.  At the top right there should be a dropdown of run start dates.

![Screen Shot 2023-01-09 at 10 09 05 AM](https://user-images.githubusercontent.com/187676/211340440-3bafa357-d712-426e-9cb4-86dac06ef563.png)
